### ISSUE TITLE
Could not load file or assembly System.Security.Cryptography.ProtectedData in .NET 6

### ISSUE THREAD

**Issue by gabrielbellussi** [dotnet/runtime#64101]
I have a solution with two projects inside, one main project to software interface and one Worker that's used as WindowsService to manage all stuffs that my solution needs to do. Both uses an Encryption method from my Utils library class and inside this method i use the ProtectedData class.

This solution was in Framework 4.7.2 and i migrated to .NET 6 recently. I performed some manual tests with the Install/Uninstall and Start/Stop of the service after the migration and all worked well. 

Yesterday i tried to run another test and got an error when my code try to use ProtectedData class from System.Security.Cryptography. 
My friend installed the same VisualStudio version (2022) and cloned the repository to check if he get the same error, and on his computer everything works well like intended to be. It's literally the same code in two different computers, on his the code works and on mine i got this error.

The error presented on my computer, when i run the service, is 
```
Application: (hided)
CoreCLR Version: 6.0.121.56705
.NET Version: 6.0.1
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IO.FileNotFoundException: Could not load file or assembly 'System.Security.Cryptography.ProtectedData, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
File name: 'System.Security.Cryptography.ProtectedData, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
(stack traces hided)
```
I need to hide the project name and the stack traces because of the organization politics, i believe that's not a big deal on the main problem. The stack trace brings to the use of ProtectedData, called inside the Service (i use ProtectedData one time on interface too, but there all works well).

Basically looks like when the WindowsService project try to use the ProtectedData, he can't find the dll. When the interface solution use the ProtectedData, everything works well. I already tried to add the reference for the System.Security.Cryptography.ProtectedData with NuGet (don't make sense, it's inside the runtime), tried to copy the .dll file to root path of the Service project, but nothing worked. All gets stranger when it work well on my friend computer and not in mine (i tried to reinstall VS and delete/clone repository again, nothing worked).

I already searched a lot about this and couldn't find anything that helps, so i want to check if anyone already had the same issue or know anything about this that can help.

**Comment by vitek-karas** [comment]
>  I already tried to add the reference for the System.Security.Cryptography.ProtectedData with NuGet (don't make sense, it's inside the runtime)

`System.Security.Cryptography.ProtectedData.dll` is not part of .NET 6 runtime. I only see it as a NuGet. So I think the interesting question would be how does it even compile if the NuGet is not referenced by the app.

Since you have a friend for whom the app works, you could try:
* Asking the friend to send you the built app (just the binaries) and try to run it on your machine
* If that works, compare the output of your build with the working version. Specifically I would diff the `app.deps.json` file as that should contain the list of all assemblies in the app.
* Produce MSBuild binlogs on both machines and compare those...

**Comment by gabrielbellussi** [comment]
@vitek-karas thanks for reply.
We found the solution!

> `System.Security.Cryptography.ProtectedData.dll` is not part of .NET 6 runtime. I only see it as a NuGet. So I think the interesting question would be how does it even compile if the NuGet is not referenced by the app.
 
Well, the ProtectedData only work for Windows and probably because of that, the framework that have it is called Microsoft.WindowsDesktop.App.

![image](https://user-images.githubusercontent.com/16911231/150784443-2a41de34-b9ca-442f-8280-da5fe47ab48c.png)

When we include this on the service project, he find the library and all work ok.
Don't make sense to me that all solution compiles without this, but at least this worked. (i can access the class with Ctrl+Click without include the framework too)

If anyone in the future is searching for the solution, we opened the .csproj and inserted this

```
<ItemGroup>
	<FrameworkReference Include="Microsoft.WindowsDesktop.App" />
</ItemGroup>
```

> Since you have a friend for whom the app works, you could try:

About that, on my friend's computer, the program never reach the process that use ProtectedData, because of our configuration, so that's why here never get the error.
